src/containers/student-home-page/find-tutor-block/FindTutorBlock.jsx

```diff
@@ -1,68 +1,61 @@
 import { useCallback, useState } from 'react'
 import { useNavigate } from 'react-router'
 import { useTranslation } from 'react-i18next'
+import { Link } from 'react-router-dom'
 
 import TitleBlock from '~/components/title-block/TitleBlock'
-import Button from '@mui/material/Button'
-import IconButton from '@mui/material/IconButton'
-import TextField from '@mui/material/TextField'
 import SearchIcon from '@mui/icons-material/Search'
 
 import useBreakpoints from '~/hooks/use-breakpoints'
 import { styles } from '~/containers/student-home-page/find-tutor-block/find-tutor-block.styles'
 import bag from '~/assets/img/student-home/bag.png'
 import { translationKey } from '~/containers/student-home-page/find-tutor-block/constants'
 import { authRoutes } from '~/router/constants/authRoutes'
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
+import AppButton from '~/components/app-button/AppButton'
 
 const FindTutorBlock = () => {
-  const [filter, setFilter] = useState('')
+  const [inputValue, setInputValue] = useState('')
   const { t } = useTranslation()
   const navigate = useNavigate()
   const { isMobile } = useBreakpoints()
+  const findOffers = `${authRoutes.findOffers.path}?search=${inputValue}`
 
   const onChange = (e) => {
-    setFilter(e.target.value)
+    setInputValue(e.target.value)
   }
 
-  const redirect = useCallback(() => {
-    return navigate(authRoutes.findOffers.path, { state: { filter } })
-  }, [filter, navigate])
-
   const handleEnterPress = useCallback(
     (e) => {
-      if (e.key === 'Enter' && filter) {
-        redirect()
+      if (e.key === 'Enter' && inputValue) {
+        navigate(authRoutes.findOffers.path, { state: { inputValue } })
       }
     },
-    [filter, redirect]
+    [inputValue, navigate]
   )
 
+  const onClear = () => setInputValue('')
+
   return (
-    <TitleBlock img={bag} translationKey={translationKey}>
-      <TextField
-        InputProps={{
-          endAdornment: (
-            <IconButton onClick={redirect}>
-              <SearchIcon position='end' />
-            </IconButton>
-          ),
-          autoComplete: 'off'
-        }}
+    <TitleBlock
+      img={bag}
+      mt={{ ...styles.root }}
+      sx={{ ...styles.img }}
```

Comment:
Why do you spread styles here? And it is bad to pass only one margin top-like style.

Reply:
fixed